framework/base/CComponent.php

```diff
@@ -609,7 +609,14 @@ public function evaluateExpression($_expression_,$_data_=array())
 		if(is_string($_expression_))
 		{
 			extract($_data_);
-			return eval('return '.$_expression_.';');
+			try
+			{
+				return @eval('return ' . $_expression_ . ';');
```

Comment:
For suppressing errors from eval() function. I think in this case we will simply parse a necessary exception or get clear false without any additional messages. Similar solution you can see \CChoiceFormat::evaluate or in the \CAuthManager::executeBizRule or \CTimestampBehavior::getTimestampByAttribute